### PR TITLE
fix(facebook); pass on facebook error message

### PR DESCRIPTION
**Pull request by doktordirk**
current error callback using new Error accessToken doens't work well when accessToken  is an object. output: {name: "Error", status: 500, message: "[object Object]",…}. not helpfull :( 

proposed change yields:
{"error":{"status":500,"message":"(#100) Tried accessing nonexisting field (nname) on node type (User)","type":"OAuthException","code":100,"fbtrace_id":"HK0ICcRDPh/"}}
better ;)

alternative to the pull request:
err = if err then err else ( if accessToken && accessToken instanceof Object && accessToken.error then  accessToken.error else null )

### CODE REVIEW

File: source/facebook/index.coffee [hunk 1]
```diff
@@ -28,7 +28,10 @@ module.exports = (server, options) ->
         debug JSON.stringify err
         return callback err
       if res.statusCode isnt 200
-        err = new Error accessToken
+        if accessToken && accessToken instanceof Object && accessToken.error
```
Comment (jdrouet) [inline source/facebook/index.coffee]: Use and instead of &&

**Comment by doktordirk** [comment]
this is actually a critical fix now, as the wrong user gets logged in if errors are not cought

File: lib/facebook/index.js [hunk 2]
```diff
@@ -37,7 +37,11 @@ module.exports = function(server, options) {
         return callback(err);
       }
       if (res.statusCode !== 200) {
-        err = new Error(accessToken);
+        if (accessToken && accessToken instanceof Object && accessToken.error) {
```
Comment (jdrouet) [inline lib/facebook/index.js]: could you use "and" instead of "&&" ?

**Comment by jdrouet** [comment]
@doktordirk thank you ;)

**Pull request merged**